models/bamf_nnunet_mr_breast/utils/NNUnetRunnerV2.py

```diff
@@ -0,0 +1,210 @@
+"""
+-------------------------------------------------
+MHub - NNU-Net Runner v2
+       Runner for pre-trained nnunet v2 models. 
+-------------------------------------------------
+
+-------------------------------------------------
+Author: Rahul Soni
+Email:  rahul.soni@bamfhealth.com
+-------------------------------------------------
+"""
+
+
+from typing import List, Optional
+import os, subprocess, shutil
+import SimpleITK as sitk, numpy as np
+from mhubio.core import Module, Instance, InstanceData, DataType, FileType, IO
+
+
+nnunet_dataset_name_regex = r"Dataset[0-9]{3}_[a-zA-Z0-9_]+"
+
+@IO.ConfigInput('in_data', 'nifti:mod=mr', the="input data to run nnunet on")
+@IO.Config('nnunet_dataset', str, None, the='nnunet dataset name')
+@IO.Config('nnunet_config', str, None, the='nnunet model name (2d, 3d_lowres, 3d_fullres, 3d_cascade_fullres)')
+@IO.Config('folds', int, None, the='number of folds to run nnunet on')
+@IO.Config('use_tta', bool, True, the='flag to enable test time augmentation')
+@IO.Config('export_prob_maps', bool, False, the='flag to export probability maps')
+@IO.Config('prob_map_segments', list, [], the='segment labels for probability maps')
+@IO.Config('roi', str, None, the='roi or comma separated list of roi the nnunet segments')
+class NNUnetRunnerV2(Module):
+
+    nnunet_dataset: str
+    nnunet_config: str
+    input_data_type: DataType
+    folds: int                          # TODO: support optional config attributes
+    use_tta: bool
+    export_prob_maps: bool
+    prob_map_segments: list
+    roi: str
+
+    def export_prob_mask(self, nnunet_out_dir: str, ref_file: InstanceData, output_dtype: str = 'float32', structure_list: Optional[List[str]] = None):
+        """
+        Convert softmax probability maps to NRRD. For simplicity, the probability maps
+        are converted by default to UInt8
+        Arguments:
+            model_output_folder : required - path to the folder where the inferred segmentation masks should be stored.
+            ref_file            : required - InstanceData object of the generated segmentation mask used as reference file.
+            output_dtype        : optional - output data type. Data type float16 is not supported by the NRRD standard,
+                                            so the choice should be between uint8, uint16 or float32. 
+            structure_list      : optional - list of the structures whose probability maps are stored in the 
+                                            first channel of the `.npz` file (output from the nnU-Net pipeline
+                                            when `export_prob_maps` is set to True). 
+        Outputs:
+            This function [...]
+        """
```

Comment:
Do we need this in this specific implementation? If not, I suggest we remove it to keep all custom modules as specific and simple to use for the end-user.

Reply:
Correct, we don't need that. Removing it